test_issue_1931_fixes.py

```
@@ -0,0 +1,81 @@
+    plan = Plan(
+        name="test_plan",
+        description="Test plan for approval"
+    )
+    
+    # Test approval callback with auto_approve=False (default)
+    callback = ApprovalCallback(auto_approve=False)
+    
+    # Since this script runs in non-interactive mode, approval should succeed
+    # even with auto_approve=False due to our stdin.isatty() check
+    result = callback(plan)
+    assert result == True, "Plan should be auto-approved in non-interactive environment"
```

Ruff (0.15.17)

[error] Avoid equality comparisons to `True`; use `result:` for truth checks

Replace with `result`

(E712)